libs/sql-parser/src/test/java/io/crate/sql/parser/TestStatementBuilder.java

```diff
@@ -2045,6 +2045,18 @@ public void test_add_column_drop_column_not_supported_in_the_same_stmt() {
             .hasMessage("line 1:30: mismatched input 'add' expecting 'DROP'");
     }
 
+    @Test
+    public void test_like_with_pattern_as_a_c_style_string() {
+        printStatement("select 'TextToMatch' LIKE E'Te\\%tch'");
+        printStatement("select 'TextToMatch' NOT LIKE E'Te\\%tch'");
+        printStatement("select 'TextToMatch' ILIKE E'te\\%tch'");
+        printStatement("select 'TextToMatch' NOT ILIKE E'te\\%tch'");
+        printStatement("select 'TextToMatch' LIKE ANY (array[E'Te\\%tch'])");
+        printStatement("select 'TextToMatch' NOT LIKE ANY (array[E'Te\\%tch'])");
+        printStatement("select 'TextToMatch' ILIKE ANY (array[E'te\\%tch'])");
+        printStatement("select 'TextToMatch' NOT ILIKE ANY (array[E'te\\%tch'])");
+    }
```

Comment:
I thought there is an issue with LIKE ANY as parser couldn't recognise such strings. But all fine, problem was in missed ) in such test cases (and copy pasting them) 🤦